File: app/src/main/java/com/geridea/trentastico/gui/views/requestloader/LessonsLoadingMessage.java
package com.geridea.trentastico.gui.views.requestloader;

import com.geridea.trentastico.utils.time.WeekInterval;

public class LessonsLoadingMessage extends AbstractTextMessage {

    private final boolean isARetry;

    private WeekInterval intervalToLoad;

    public LessonsLoadingMessage(int operationId, WeekInterval intervalToLoad, boolean isARetry) {
        super(operationId);

        this.intervalToLoad = intervalToLoad;
        this.isARetry = isARetry;
    }

    @Override
    public String getText() {
        String fromTo = formatFromToString(intervalToLoad);

        return isARetry ?
                "Sto riprovando a scaricare gli orari dal "+fromTo+"...":
                "Sto scaricando gli orari dal "+fromTo+"...";
    }

}

File: app/src/main/java/com/geridea/trentastico/gui/views/requestloader/LessonsLoadingMessage.kt
package com.geridea.trentastico.gui.views.requestloader

class LessonsLoadingMessage(operationId: Int, private val isARetry: Boolean):
        AbstractTextMessage(operationId) {

    override val text: String
        get() {
            return if (isARetry)
                "Sto riprovando a scaricare gli orari del tuo corso di studi..."
            else
                "Sto scaricando gli orari del tuo corso di studi..."
        }

}
